ml-agents/mlagents/trainers/trainer/rl_trainer.py

```diff
@@ -80,6 +81,41 @@ def _is_ready_update(self):
         """
         return False
 
+    def _policy_mean_reward(self) -> Optional[float]:
+        """ Returns the mean episode reward for the current policy. """
+        rewards = self.cumulative_returns_since_policy_update
+        if len(rewards) == 0:
+            return None
+        else:
+            return sum(rewards) / len(rewards)
+
+    @timed
+    def _checkpoint(self) -> None:
+        """
+        Saves the model
+        """
+        n_policies = len(self.policies.keys())
+        if n_policies > 1:
+            logger.warning(
+                "Trainer has multiple policies, but default behavior only saves the first."
+            )
+        policy = list(self.policies.values())[0]
+        policy.checkpoint(self._policy_mean_reward())
+
+    def save_model(self) -> None:
+        """
+        Exports the model
+        """
+        n_policies = len(self.policies.keys())
+        if n_policies > 1:
+            logger.warning(
+                "Trainer has multiple policies, but default behavior only saves the first."
+            )
+        policy = list(self.policies.values())[0]
+        policy_mean_reward = self._policy_mean_reward()
+        policy.checkpoint(policy_mean_reward)
+        policy.save(policy_mean_reward)
```

Comment:
Shouldn't `save` call `checkpoint` as well?

Reply:
Keeping them together was getting confusing and rather nested, checkpoint saving and final model saving is done separately. Thats why in the save_model, there is an explicit call to saving a checkpoint (line 116) and a separate call to save on line 117.